Hydra/WrapperStream.cs

```diff
@@ -8,10 +8,12 @@ namespace Hydra
     internal abstract class WrapperStream : ReadOnlyStream
     {
         private readonly Stream stream;
+        private readonly bool ownStream;
 
-        protected WrapperStream(Stream stream)
+        protected WrapperStream(Stream stream, bool ownStream = false)
```

Comment:
I'd prefer if this was not a default parameter, since there's no real reason one should be the default over the other in which case the behaviour should probably always be specified by the caller.